apps/shopify/src/dataTransformer.js

```diff
@@ -1,9 +1,29 @@
 import get from 'lodash/get';
-import last from 'lodash/last';
 import flatten from 'lodash/flatten';
 import { DEFAULT_SHOPIFY_VARIANT_TITLE } from './constants';
 import { convertBase64ToString, convertStringToBase64 } from './utils/base64';
 
+/**
+ * Decodes the ID of a Shopify resource
+ * The ID is encoded in base64 and the actual ID is the last part of the string
+ * e.g. gid://shopify/Product/1234567890 -> 1234567890
+ */
+export const decodeId = (sku) => {
+  try {
+    const decodedId = convertBase64ToString(sku);
+    return decodedId && decodeId.slice(decodedId.lastIndexOf('/') + 1);
+  } catch {
+    return null;
```

Comment:
Try/catch is a good practice, but just returning the `null` in the catch is a bit of a code smell to me. Is `convertBase64ToString()` a BE call that could fail? If not, why not return line 14 without the try/catch?

Reply:
This was to keep sentiment with the old code which used a try/catch for this. `convertBase64ToString()` uses btoa which _could_ throw an error given it's handed a non-latin symbol.

I'm cool with just removing the try/catch, Shopify probably will only send us latin characters

Reply:
➕ 

I might rather have specific guardrails here around an empty `sku`, or a poorly-formatted one (not ever sure if we know what that would be)

Reply:
Tested it locally with an empty sku/malformed, and looks like it handles it gracefully and the results still load successfully